src/CodeGen_Xtensa.cpp

```diff
@@ -123,186 +127,25 @@ void CodeGen_Xtensa::compile(const Module &module) {
-            // Emit the body
-            print(body);
-            // stream << "printf(\"[end]" << simple_name << "\\n\");";
-
-            // Return success.
-            stream << get_indent() << "return 0;\n";
-            cache.clear();
-        }
-
-        // Ensure we use open/close_scope, so that the cache doesn't try to linger
-        // across function boundaries for internal closures.
-        close_scope("");
-    }
-
-    // Workaround for https://github.com/halide/Halide/issues/635:
-    // For historical reasons, Halide-generated AOT code
-    // defines user_context as `void const*`, but expects all
-    // define_extern code with user_context usage to use `void *`. This
-    // usually isn't an issue, but if both the caller and callee of the
-    // pass a user_context, *and* c_plus_plus_name_mangling is enabled,
-    // we get link errors because of this dichotomy. Fixing this
-    // "correctly" (ie so that everything always uses identical types for
-    // user_context in all cases) will require a *lot* of downstream
-    // churn (see https://github.com/halide/Halide/issues/7298),
-    // so this is a workaround: Add a wrapper with `void*`
-    // ucon -> `void const*` ucon. In most cases this will be ignored
-    // (and probably dead-stripped), but in these cases it's critical.
-    //
-    // (Note that we don't check to see if c_plus_plus_name_mangling is
-    // enabled, since that would have to be done on the caller side, and
-    // this is purely a callee-side fix.)
-    if (f.linkage != LinkageType::Internal &&
-        output_kind == CPlusPlusImplementation &&
-        target.has_feature(Target::CPlusPlusMangling) &&
-        get_target().has_feature(Target::UserContext)) {
-
-        Type ucon_type = Type();
-        for (const auto &arg : args) {
-            if (arg.name == "__user_context") {
-                ucon_type = arg.type;
-                break;
-            }
-        }
-        if (ucon_type == type_of<void const *>()) {
-            stream << "\nHALIDE_FUNCTION_ATTRS\n";
-            stream << "int " << simple_name << "(";
-            emit_arg_decls(type_of<void *>());
-            stream << ") ";
-            open_scope();
-            stream << get_indent() << "    return " << simple_name << "(";
-            const char *comma = "";
-            for (const auto &arg : args) {
-                if (arg.name == "__user_context") {
-                    // Add an explicit cast here so we won't call ourselves into oblivion
-                    stream << "(void const *)";
-                }
-                stream << comma << print_name(arg.name);
-                if (arg.is_buffer()) {
-                    stream << "_buffer";
-                }
-                comma = ", ";
-            }
-            stream << ");\n";
-            close_scope("");
-        }
-    }
+void CodeGen_Xtensa::compile(const LoweredFunc &func, const MetadataNameMap &metadata_name_map) {
```

Comment:
Why is still needed?

Reply:
Without it, we get

```
/Users/srj/GitHub/Halide/src/CodeGen_Xtensa.h:21:10: warning: 'Halide::Internal::CodeGen_Xtensa::compile' hides overloaded virtual function [-Woverloaded-virtual]
    void compile(const Module &module);
         ^
/Users/srj/GitHub/Halide/src/CodeGen_C.h:69:18: note: hidden overloaded virtual function 'Halide::Internal::CodeGen_C::compile' declared here: different number of parameters (2 vs 1)
    virtual void compile(const LoweredFunc &func, const MetadataNameMap &metadata_name_map);
```

(arguably, we should rename these to avoid overloading, to avoid this nonsense -- I may propose that as a followup)